airgun/views/computeresource.py

```diff
@@ -199,34 +218,207 @@ class compute_profiles(SatTab):
+
+    provider_content = ConditionalSwitchableView()
 
     submit = Text('//input[@name="commit"]')
 
+    @property
+    def current_provider(self):
+        # Return a widget to be read by ConditionalSwitchableView instance and to be passed to
+        # _is_provider function.
+        return self.compute_resource
+
+    @provider_content.register(functools.partial(_is_provider, 'EC2'))
+    class EC2ResourceForm(View):
+        flavor = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_flavor_id')
+        image = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_image_id')
+        availability_zone = FilteredDropdown(
+            id='s2id_compute_attribute_vm_attrs_availability_zone')
+        subnet = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_subnet_id')
+        security_groups = MultiSelect(id='ms-compute_attribute_vm_attrs_security_group_ids')
+        managed_ip = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_managed_ip')
+
+    @provider_content.register(functools.partial(_is_provider, 'RHV'))
+    class RHVResourceForm(View):
+        cluster = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_cluster')
+        template = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_template')
+        instance_type = FilteredDropdown(id='s2id_compute_attribute_vm_attrs_instance_type')
+        cores = TextInput(id='compute_attribute_vm_attrs_cores')
+        sockets = TextInput(id='compute_attribute_vm_attrs_sockets')
+        memory = TextInput(id='compute_attribute_vm_attrs_memory')
+        highly_available = Checkbox(id='compute_attribute_vm_attrs_ha')
+
+        @View.nested
+        class network_interfaces(RemovableWidgetsItemsListView):
+            ROOT = "//fieldset[@id='network_interfaces']"
+            ITEM_WIDGET_CLASS = ComputeResourceRHVProfileNetworkItem
+
+        @View.nested
+        class storage(RemovableWidgetsItemsListView):
+            ROOT = "//fieldset[@id='storage_volumes']"
+            ITEMS = "./div/div[contains(@class, 'removable-item')]"
+            ITEM_WIDGET_CLASS = ComputeResourceRHVProfileStorageItem
+
     @property
     def is_displayed(self):
         breadcrumb_loaded = self.browser.wait_for_element(
-            self.breadcrumb, exception=False)
+            self.breadcrumb, exception=False, ensure_page_safe=True, timeout=10)
+        if not breadcrumb_loaded:
+            return False
+        breadcrumb_text = self.breadcrumb.read()
         return (
-            breadcrumb_loaded
-            and self.breadcrumb.locations[0] == 'Compute resources'
-            and self.breadcrumb.locations[2] == 'Compute profiles'
-            and self.breadcrumb.read().startswith('Edit ')
+                self.breadcrumb.locations[0] == 'Compute Resources'
+                and self.breadcrumb.locations[2] == 'Compute Profiles'
+                and (breadcrumb_text.startswith('Edit ') or breadcrumb_text.startswith('New '))
```

Comment:
any reason to have both views in one class? Usually we differentiate them

Reply:
This is not my implementation, I only fixing here, but if you want can split them.

Reply:
But this for is a bit tricky as we are not creating or editing, but associating

Reply:
it is not critical